apps/api/src/app/step-schemas/usecases/get-step-schema/get-step-schema.command.ts

```diff
@@ -1,10 +1,23 @@
 import { EnvironmentWithUserCommand } from '@novu/application-generic';
 import { ActionStepEnum, ChannelStepEnum, StepType } from '@novu/framework';
-import { IsEnum } from 'class-validator';
+import { IsEnum, IsNotEmpty, IsString } from 'class-validator';
 
 const StepTypeValue = { ...ChannelStepEnum, ...ActionStepEnum };
 
-export class GetStepSchemaCommand extends EnvironmentWithUserCommand {
+export class GetStepTypeSchemaCommand extends EnvironmentWithUserCommand {
   @IsEnum(StepTypeValue)
+  @IsNotEmpty()
   stepType: StepType;
 }
+
+export class GetExistingStepSchemaCommand extends EnvironmentWithUserCommand {
```

Comment:
Nitpicky comment: I'd use one command to make all fields optional to avoid an extra [if here](https://github.com/novuhq/novu/pull/6651/files#diff-cbaa2172c5a107d6e34e4ef3c67d70200da6697594a666ce8ff9b2cfb924adffR34). You can safely ignore this comment,

Reply:
I won't lie, i was not sure about the solution. I structured it this way for two reasons: first, I wanted to pass the validation to `class-validator`, and second, I wanted to make sure each command had a clear interface to keep the code cleaner (in a way).